支持提取 Quilt 崩溃堆栈中包含的 Mod 信息

### 检查项

- [x] 我已在 [Issues 页面](https://github.com/PCL-Community/PCL2-CE/issues?q=is%3Aissue+) 中搜索，确认了这一 Bug 未被提交过。
- [x]  我已查看 [功能投票页面](https://github.com/Hex-Dragon/PCL2/discussions/categories/%E5%8A%9F%E8%83%BD%E6%8A%95%E7%A5%A8/)，确认了这一提案未在投票列表中。

### 描述

Quilt 加载器的实例在崩溃时，无法准确分析是由哪个 Mod 导致的，参见下方日志

![Image](https://github.com/user-attachments/assets/969ed57c-a041-4ff7-b4ef-116a989a147f)

```
#Ln 411 起
Caused by: java.lang.NoClassDefFoundError: net/irisshaders/iris/gui/screen/ShaderPackScreen
	at dev.isxander.xso.XandersSodiumOptions.convertCategory(XandersSodiumOptions.java:113) ~[xanders-sodium-options-2.2.11.20.1.i0:0/:?]
	at dev.isxander.xso.XandersSodiumOptions.wrapSodiumScreen(XandersSodiumOptions.java:52) ~[xanders-sodium-options-2.2.11.20.1.i0:0/:?]
	at net.minecraft.class_310.localvar$cel000$xanders-sodium-options$modifyScreen(class_310.java:14013) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_310.method_1507(class_310.java) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_429.method_47624(class_429.java:153) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_4185.method_25306(class_4185.java:94) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_4264.method_25348(class_4264.java:56) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_339.method_25402(class_339.java:189) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_4069.method_25402(class_4069.java:38) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_312.method_1611(class_312.java:98) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_437.method_25412(class_437.java:409) ~[minecraft-1.20.1.i0:0/:?]
	... 24 more
```
实际上是 xanders-sodium-options-2.2.11.20.1.jar 的问题


### 日志与附件

[错误报告-2025-1-24_上午_08.06.59.zip](https://github.com/user-attachments/files/18529393/-2025-1-24_._08.06.59.zip)

--- a/Plain Craft Launcher 2/Modules/Minecraft/ModCrash.vb
+++ b/Plain Craft Launcher 2/Modules/Minecraft/ModCrash.vb
@@ -405,6 +405,14 @@ Extracted:
         _version = version
         Log("[Crash] 步骤 3：分析崩溃原因")
         LogAll = If(LogMc, If(LogMcDebug, "")) & If(LogHs, "") & If(LogCrash, "")
+        
+        '处理 Quilt Mod Table 以避免错误分析 (CE #107)
+        If LogAll.Contains("quilt") AndAlso LogAll.Contains("Mod Table Version") Then
+            Log("[Crash] 处理 Quilt Mod Table 后再继续分析")
+            Dim beforeTable = LogAll.BeforeFirst("| Index")
+            Dim afterTable = LogAll.AfterFirst("Mod Table Version:")
+            LogAll = beforeTable & afterTable
+        End If
 
         '1. 精准日志匹配，中/高优先级
         AnalyzeCrit1()
@@ -698,7 +706,7 @@ Done:
             'If Not Stack.Contains(".") Then Continue For
             For Each IgnoreStack In {
                 "java", "sun", "javax", "jdk", "oolloo",
-                "org.lwjgl", "com.sun", "net.minecraftforge", "paulscode.sound", "com.mojang", "net.minecraft", "cpw.mods", "com.google", "org.apache", "org.spongepowered", "net.fabricmc", "com.mumfrey",
+                "org.lwjgl", "com.sun", "net.minecraftforge", "paulscode.sound", "com.mojang", "net.minecraft", "cpw.mods", "com.google", "org.apache", "org.spongepowered", "net.fabricmc", "com.mumfrey", "org.quiltmc",
                 "com.electronwill.nightconfig", "it.unimi.dsi",
                 "MojangTricksIntelDriversForPerformance_javaw"}
                 If Stack.StartsWithF(IgnoreStack) Then GoTo NextStack
@@ -707,6 +715,7 @@ Done:
 NextStack:
         Next
         PossibleStacks = PossibleStacks.Distinct.ToList
+        
         Log("[Crash] 找到 " & PossibleStacks.Count & " 条可能的堆栈信息")
         If Not PossibleStacks.Any() Then Return New List(Of String)
         For Each Stack As String In PossibleStacks
@@ -766,6 +775,11 @@ NextStack:
             If IsFabricDetail Then
                 Details = Details.Replace("Fabric Mods", "¨")
                 Log("[Crash] 崩溃报告中检测到 Fabric Mod 信息格式")
+            End If
+            Dim isQuiltDetail As Boolean = Details.Contains("quilt-loader")
+            If isQuiltDetail Then
+                Details = Details.Replace("Mod Table Version", "¨")
+                Log("[Crash] 崩溃报告中检测到 Quilt Mod 信息格式")
             End If
             Details = Details.AfterLast("¨")
 

